HelloImKevo/AndroidPresentationDisplayDemo
# Add modified Carousel view from AndroidX Constraint lib to cycle thru static ads

## Description:
After experimenting with the native `Carousel` class, I was stumped by a bug with the infinite transition mechanism, when automating it with a timer task scheduled at fixed intervals. I narrowed down the root cause to a needed one-liner change in the native class. But since all of the fields are `private` I could not extend and override the behavior; so I just copied the entire class and made the needed fix. Maybe I'll open up a bugfix in the AOSP repo; this change has some minor side effects on other Carousel modes of operation, so it'll require more of a time sink.

## Test Plan:
- Launch the app.
- Click on the Carousel option in the side navigation drawer.
- Confirm that the carousel images load, and start their automatic slide transitions when the fragment is resumed.
- Confirm that the Carousel infinite loop mechanism works correctly.
- Confirm that no crashes occur.

## Related Issue(s):
https://gpproduct.atlassian.net/browse/PPA-9661

## Live Action:
![carousel-screenshot-01](https://github.com/HelloImKevo/AndroidPresentationDisplayDemo/assets/4968350/5b1b4e1a-cb31-4f44-ae87-c848f10846fd)
